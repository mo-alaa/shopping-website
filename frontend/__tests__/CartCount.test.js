import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import CartCount from "../components/CartCount";

describe("<Cartount/>", () => {
  xit("renders", () => {
    shallow(<CartCount count={10} />);
  });

  xit("matches the snapshot", () => {
    const wrapper = shallow(<CartCount count={12} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  xit("updates via props", () => {
    //while shallow goes only 1 level deep
    const wrapper = shallow(<CartCount count={50} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
    wrapper.setProps({ count: 7 });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  xit("updates via props", () => {
     //mount actually run it in like sort of a browser environment
     //we usually will be using mount, because most of our components
     //are not as clean as cartcount
       //and we need our test to be as close to the real experience
     const wrapper = mount(<CartCount count={50} />);
    //  console.log(wrapper.debug());
 
  });
});
