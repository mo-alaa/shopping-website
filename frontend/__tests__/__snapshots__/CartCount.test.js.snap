// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cartount/> matches the snapshot 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="12"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        12
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`<Cartount/> updates via props 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="50"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        50
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`<Cartount/> updates via props 2`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="7"
      timeout={
        Object {
          "enter": 400,
          "exit": 400,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        7
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;
